dbms/src/Functions/array/arrayCompact.cpp

```diff
@@ -0,0 +1,141 @@
+#include <DataTypes/DataTypesNumber.h>
+#include <Columns/ColumnsNumber.h>
+#include "FunctionArrayMapped.h"
+#include <Functions/FunctionFactory.h>
+
+
+namespace DB
+{
+/// arrayCompact(['a', 'a', 'b', 'b', 'a']) = ['a', 'b', 'a'] - compact arrays
+    namespace ErrorCodes
+    {
+        extern const int ILLEGAL_COLUMN;
+    }
+
+    struct ArrayCompactImpl
+    {
+        static bool needBoolean() { return false; }
+        static bool needExpression() { return false; }
+        static bool needOneArray() { return false; }
+
+        static DataTypePtr getReturnType(const DataTypePtr & expression_return, const DataTypePtr & /*array_element*/)
+        {
+            WhichDataType which(expression_return);
+
+            if (which.isNativeUInt())
+                return std::make_shared<DataTypeArray>(std::make_shared<DataTypeUInt64>());
+
+            if (which.isNativeInt())
+                return std::make_shared<DataTypeArray>(std::make_shared<DataTypeInt64>());
+
+            if (which.isFloat())
+                return std::make_shared<DataTypeArray>(std::make_shared<DataTypeFloat64>());
+
+            throw Exception("arrayCompact cannot add values of type " + expression_return->getName(), ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+        }
+
+
+        template <typename Element, typename Result>
+        static bool executeType(const ColumnPtr & mapped, const ColumnArray & array, ColumnPtr & res_ptr)
+        {
+            const ColumnVector<Element> * column = checkAndGetColumn<ColumnVector<Element>>(&*mapped);
+
+            if (!column)
+            {
+                const ColumnConst * column_const = checkAndGetColumnConst<ColumnVector<Element>>(&*mapped);
```

Comment:
Looks like two functions in one. Maybe `IFunction::useDefaultImplementationForConstants` can help you? If you override it to return true, the function executor will cast away const automatically, and you will only have to provide the non-const implementation.

Reply:
It works well. Thanks! BTW, what should I do to pass the stress test?